Sources/Crypto/Cipher/OpenSSLStreamCipher.swift

```diff
@@ -0,0 +1,135 @@
+import CNIOOpenSSL
+import Foundation
+import Bits
+
+/// Available cipher modes. Either `encrypt` or `decrypt`.
+///
+/// Used when calling `reset` on a `Cipher`.
+public enum CipherMode: Int32 {
+    /// Encrypts arbitrary data to encrypted ciphertext.
+    case encrypt = 1
+
+    /// Decrypts encrypted ciphertext back to its original value.
+    case decrypt = 0
+}
+
+/// Wrapper to allow for safely working with a potentially-nil Data's byte buffer.
+extension Optional where Wrapped == Data {
+    // Note: It's iffy to try this with a mutable buffer, so an Optional version
+    // of withMutableByteBuffer is not provided.
+    func withByteBuffer<T>(_ closure: (BytesBufferPointer?) throws -> T) rethrows -> T {
+        switch self {
+        case .some(let data):
+            return try data.withByteBuffer({ try closure($0) })
+        case .none:
+            return try closure(nil)
+        }
+    }
+}
+
+/// OpenSSLStreamCipher is a protocol representing a higher-level interface for managing various OpenSSL stream ciphers.
+public protocol OpenSSLStreamCipher {
+    func reset(key: LosslessDataConvertible, iv: LosslessDataConvertible?, mode: CipherMode) throws
```

Comment:
All protocol methods must be documented (this counts toward the overall % documented for the framework)

Reply:
Sounds good - will do shortly.